sympy/printing/tests/test_c.py

```diff
@@ -699,8 +699,8 @@ def check(expr, ref):
         check(gamma(x), 'tgamma{s}(x)')
         check(loggamma(x), 'lgamma{s}(x)')
 
-        check(ceiling(x + 2.), "ceil{s}(x + 2.0{S})")
-        check(floor(x + 2.), "floor{s}(x + 2.0{S})")
+        check(ceiling(x + 2.), "ceil{s}(x) + 2.0{S}")
+        check(floor(x + 2.), "floor{s}(x) + 2.0{S}")
```

Comment:
This is potentially problematic, as I noted on the other PR, especially for the purposes of codegen. These two expressions aren't actually equal for floating point numbers. 

(but then again, SymPy does a lot of simplifications that aren't true in general for floating-point numbers)

Reply:
SymPy is returning a result that is `symbolically` correct, even if `x = Float(1e-16)`. It seems like that is in line with SymPy's mission. Guessing whether someone wants an expression to suffer truncation or not is not, so I think making these consistent is the way to go.  Please see #25872 for just this addition. And I think I prefer #25875 for the more robust and less controversial change to handle #23731.